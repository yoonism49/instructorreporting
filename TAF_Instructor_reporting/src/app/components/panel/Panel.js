// @flow weak

import React      from 'react';
import PropTypes  from 'prop-types';

const Panel = ({
  hasTitle,
  title,
  bodyBackGndColor,
  bodyCustomClass,
  sectionCustomClass,
  children
}) => (
  <section className={`panel ${sectionCustomClass}`}>
    {
      hasTitle &&
      <header className="panel-heading">
        {title}
      </header>
    }
    <div
      className={`panel-body ${bodyCustomClass}`}
      style={{backgroundColor: bodyBackGndColor}}>
      { children }
    </div>
  </section>
);

Panel.propTypes = {
  hasTitle:           PropTypes.bool,
  title:              PropTypes.string,
  bodyBackGndColor:   PropTypes.string,
  bodyCustomClass:    PropTypes.string,
  sectionCustomClass: PropTypes.string,
  children:           PropTypes.node
};

Panel.defaultProps = {
  hasTitle:           true,
  title:              'no title set',
  sectionCustomClass: '',
  bodyBackGndColor:   '#FFF'
};

export default Panel;
