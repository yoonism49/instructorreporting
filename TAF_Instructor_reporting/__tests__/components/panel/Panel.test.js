
'use strict';

import React              from 'react';
import Panel              from '../../../src/app/components/panel/Panel';
import renderer       from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme

describe('Panel component', () => {
  it('renders as expected', () => {
    const mockProps = {
      hasTitle: true,
      title: 'a title',
      bodyBackGndColor: '#F1F2F3',
      bodyCustomClass: '',
      sectionCustomClass: ''
    };
    const component = renderer.create(
      <div>
        <Panel {...mockProps }>
          <p>
            panel children
          </p>
        </Panel>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
