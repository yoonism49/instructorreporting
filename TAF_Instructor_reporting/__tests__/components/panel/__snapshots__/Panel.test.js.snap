// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel component renders as expected 1`] = `
<div>
  <section
    className="panel "
  >
    <header
      className="panel-heading"
    >
      a title
    </header>
    <div
      className="panel-body "
      style={
        Object {
          "backgroundColor": "#F1F2F3",
        }
      }
    >
      <p>
        panel children
      </p>
    </div>
  </section>
</div>
`;
